swaggen/src/main/java/utils/io/FileMapper.java

```diff
@@ -37,15 +47,79 @@
 	 * Maps a Java object to a YAML file.
 	 * 
 	 * @param filename the name of the YAML file
-	 * @param object the Java object
+	 * @param newSwagger the Java object, typically swagger object 
+	 * @param <T> The type
+	 * 
+	 * @throws JsonGenerationException JsonGenerationException
+	 * @throws JsonMappingException JsonMappingException
+	 * @throws IOException IOException
+	 */
+	public static <T> void classToYaml(String filename, T newSwagger)
+			throws JsonGenerationException, JsonMappingException, IOException {
+		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+		File file = new File(filename);
+
+		checkExistingEndpoints(newSwagger, file, objectMapper);
+
+		if (newSwagger instanceof Swagger) {
+			Swagger toYaml = convertSwagger(newSwagger, filename, file, objectMapper);
+			objectMapper.writeValue(file, toYaml);
+		} else {
+			objectMapper.writeValue(file, newSwagger);
+		}
+	}
+
+	/**
+	 * Checks for existing endpoints in yaml files and if matched, add the existing
+	 * request methods to the new Swagger file.
+	 * @param swagger the swagger object containing all the endpoitns
+	 * @param file file object, containing where it is supposed to be placed
+	 * @param objectMapper the ObjectMapper
 	 * @throws JsonGenerationException
 	 * @throws JsonMappingException
 	 * @throws IOException
 	 */
-	public static <T> void classToYaml(String filename, T object)
-			throws JsonGenerationException, JsonMappingException, IOException {
-		ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
-		objectMapper.writeValue(new File(filename), object);
+	private static <T> void checkExistingEndpoints(T swagger, File file, ObjectMapper objectMapper) 

```

Comment:
A few things here.

1. Whenever you I see `instanceof` I feel like there is likely something unsafe going on. We should know the class. I can't tell just from this PR why we wouldn't.
2. We are typecasting swagger multiple times. That is wasteful.
3. Again, we can flatten this out by making use of map functionality. If we want to run this through Sonarqube eventually, this needs to be resolved. The cognitive complexity of this method is way too high. https://www.sonarsource.com/resources/white-papers/cognitive-complexity.html

Reply:
The only case where T is not Swagger is in the test testYamlWriter at src/test/java/tests/yaml

Reply:
That should indicate a problem. `testYamlWriter` should be testing that we can convert an arbitrary class to YAML. This is likely related to other comments that I left about the YAML writer not caring what is passed in. Don't have the YAML writer do any sort of reading/validating of YAML or POJOs.